### ISSUE TITLE
False positive with `prefer_self_in_static_references` rule when used inside `#keyPath`.

### ISSUE THREAD

**Issue by guyniv** [realm/SwiftLint#4489]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`prefer_self_in_static_references` opt-in rule warns on this chunk of code:

```swift
@objc class Foo: NSObject {
  @objc private var hello: String

  override init() {
    self.hello = "hello"
  }

  private func foo() {
    let keyPath = #keyPath(Foo.hello) // SwiftLint warns here.
    print("## \(keyPath)")
  }
}

```

When changing
```swift
let keyPath = #keyPath(Foo.hello)
```
to
```swift
let keyPath = #keyPath(Self.hello)
```
we get this compilation error:
```
Cannot find 'Self' in scope
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

```
Linting Swift files at paths /Users/guy/Video/Video/Video/StockAudioEpidemicViewModel.swift
Linting 'StockAudioEpidemicViewModel.swift' (1/1)
/Users/guy/Video/Video/Video/StockAudioEpidemicViewModel.swift:256:28: warning: Prefer Self in Static References Violation: Static references should be prefixed by `Self` instead of the class name. (prefer_self_in_static_references)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.49.1`.
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `Homebrew`.
* Paste your configuration file:

```yml
opt_in_rules:
  - prefer_self_in_static_references
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? `Xcode 13.3.1 Build version 13E500a`.
